.gitlab/build.yml

```diff
@@ -103,124 +57,98 @@ calculate-build-metadata:
 
-publish-internal-adp-image:
+# Finally, we publish our internal images after running through a small build process to add some necessary tooling
+# required for the images to be deployed/used internally.
+#
+# These takes them from the sort of temporary holding ground of `registry.ddbuild.io/saluki/agent-data-plane`, adds the
+# necessary pieces, and then publishes them to `registry.ddbuild.io/agent-data-plane` as a first-class citizen.
+#
+# We specifically duplicate our `ADP_INTERNAL_IMAGE_TAG` (and FIPS-specific variant) tag and construct the same value
+# manually because otherwise, the variable is evaluated in the downstream job, leading to the wrong values being
+# substituted.
+publish-adp-image-internal:
   stage: build
   extends: [.build-common-variables]
   needs:
-    - build-adp-image
+    - build-adp-image-internal
   trigger:
     project: DataDog/images
     branch: master
     strategy: depend
   variables:
-    # We always build off of the current ADP image for this commit SHA, which is our non-FIPS "debug" build.
     IMAGE_NAME: agent-data-plane
     IMAGE_VERSION: tmpl-v1
     TMPL_SRC_REPO: ${ADP_IMAGE_REPO_NAME}
-    TMPL_SRC_IMAGE: ${CI_COMMIT_SHORT_SHA}
+    TMPL_SRC_IMAGE: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}"
     RELEASE_TAG: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}"
-    BUILD_TAG: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}"
+    BUILD_TAG: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-build"
     RELEASE_STAGING: "true"
     RELEASE_PROD: "true"
 
-upload-adp-symbols:
-  extends: [.build-common-variables, .upload-adp-symbols-definition]
-  needs:
-    - build-adp-image
-  variables:
-    IMAGE_TAG: ${ADP_DEBUG_IMAGE_TAG}
-
-upload-adp-symbols-fips:
-  extends: [.build-common-variables, .upload-adp-symbols-definition]
-  needs:
-    - build-adp-image-fips
-  variables:
-    IMAGE_TAG: ${ADP_DEBUG_IMAGE_TAG_FIPS}
-
-upload-adp-release-symbols:
-  extends: [.build-common-variables, .upload-adp-symbols-definition]
-  rules:
-    - if: !reference [.on_official_release, rules, if]
-  needs:
-    - build-adp-image-release
-  variables:
-    IMAGE_TAG: ${ADP_RELEASE_IMAGE_TAG}
-
-upload-adp-release-symbols-fips:
-  extends: [.build-common-variables, .upload-adp-symbols-definition]
-  rules:
-    - if: !reference [.on_official_release, rules, if]
+publish-adp-image-internal-fips:
+  stage: build
+  extends: [.build-common-variables]
   needs:
-    - build-adp-image-release-fips
+    - build-adp-image-internal-fips
+  trigger:
+    project: DataDog/images
+    branch: master
+    strategy: depend
   variables:
-    IMAGE_TAG: ${ADP_RELEASE_IMAGE_TAG_FIPS}
+    IMAGE_NAME: agent-data-plane
+    IMAGE_VERSION: tmpl-v1
+    TMPL_SRC_REPO: ${ADP_IMAGE_REPO_NAME}
+    TMPL_SRC_IMAGE: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-fips"
+    RELEASE_TAG: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-fips"
+    BUILD_TAG: "v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-fips-build"
```

Comment:
```suggestion
    TMPL_SRC_IMAGE: "${ADP_INTERNAL_IMAGE_TAG_FIPS}"
    RELEASE_TAG: "${ADP_INTERNAL_IMAGE_TAG_FIPS}"
    BUILD_TAG: "${ADP_INTERNAL_IMAGE_TAG_FIPS}-build"
```

Reply:
We actually _can't_ deduplicate it in this way, because it leads the CI pipeline ID/commit SHA from the _downstream_ being used instead of _our_ pipeline.